test/addons/openssl-test-engine/test.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+const common = require('../../common');
+
+// This tests crypto.setEngine().
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+
+const engine = path.join(__dirname,
+                         `/build/${common.buildType}/testsetengine.engine`);
+
+if (!fs.existsSync(engine))
+  common.skip('no engine');
```

Comment:
```suggestion
```
Let's move this right above the place where we try to load the shared library, so that we don't miss out on running the tests below on platforms where the engine isn't built, like Windows, because these tests don't depend on the engine.